### ISSUE TITLE
Mismatch in results of TensorRT 8.2 & onnx-runtime

### ISSUE THREAD

**Issue by prashant-dn** [NVIDIA/TensorRT#3279]
## Description
We trained a custom classifier model based on Resnet50 + focal loss for 8 classes; obtained the onnx using 
```python3
with torch.no_grad():
    # move model to gpu
    model.to(device)
    #filepath = "resnet50.onnx"
    img_tensor,  _ = next(iter(pplDataModule.val_dataloader()))
    
    # move tensors to gpu space
    img_tensor = img_tensor.to(device)
    
    print(f"img tensor shape: {img_tensor.shape}")
    torch.onnx.export(model,                     
                      (img_tensor,), 
                      onnx_file,                  # where to save the model (can be a file or file-like object)
                      export_params=True,      
                      opset_version=11,          # the ONNX version to export the model to
                      do_constant_folding=True,  # whether to execute constant folding for optimization
                      input_names = ['image'],   # the model's input names
                      output_names = ['output'], # the model's output names
                      dynamic_axes={'image' : {0 : 'batch_size'},
                                    'output' : {0 : 'batch_size'}
                                   }
                )
    # Checks
    import onnx
    model_onnx = onnx.load(onnx_file)  # load onnx model
    onnx.checker.check_model(model_onnx)  # check onnx model

    # Simplify
    if simplify:
        try:
            cuda = torch.cuda.is_available()
            import onnxsim
            print(f'simplifying with onnx-simplifier {onnxsim.__version__}...')
            model_onnx, check = onnxsim.simplify(model_onnx)
            assert check, 'assert check failed'
            onnx.save(model_onnx, onnx_file)
            print(f'Exported simplified onnx: {onnx_file}')
        except Exception as e:
            print(f'simplifier failure: {e}')
```

I can infer on this onnx using onnx-runtime and obtain satisfactory classification results
```console
GT: key | predConf: 0.3700000047683716,  class:Keys
GT: key | predConf: 0.3400000035762787,  class:Other
GT: key | predConf: 0.2800000011920929,  class:Purse
GT: mobile | predConf: 0.9100000262260437,  class:Mobile
GT: mobile | predConf: 0.8700000047683716,  class:Mobile
GT: mobile | predConf: 0.75,  class:Mobile
```

Then I export the engine file using 
```console
trtexec --onnx=resnet50.onnx --workspace=8144  --saveEngine=resnet50-noargs.engine
```

When I infer on it using python-tensorrt, I get really low confidence scores for false classifications.
```console
GT: key1 | conf: 0.18082687258720398  pred class: Keys
GT: key2 | conf: 0.17109805345535278  pred class: QR_Code
GT: key3.| conf: 0.17180819809436798  pred class: Mobile
GT: mobile1 | conf: 0.17762377858161926  pred class: Keys
GT: mobile2 | conf: 0.1849362701177597  pred class: Keys
GT: mobile3. | conf: 0.1679219752550125  pred class: Keys
```

I have made the pre-processing for both these runtimes exactly same
```
def preprocess(img):
    # transform image
    mean_arr = (0.485, 0.456, 0.406)
    std_arr = (0.229, 0.224, 0.225)

    transform = transforms.Compose(
        [
            transforms.Resize((224, 224)),
            transforms.ToTensor(),
            transforms.Normalize(mean_arr, std_arr),
        ]
    )
    transformed_image = transform(img)
    transformed_image = torch.unsqueeze(transformed_image, 0)
    pro_img = transformed_image.detach().cpu().numpy()
    return pro_img
```

But I am unable to figure out why the results differ.

## Environment

**TensorRT Version**: 8.2.03

**NVIDIA GPU**: T4

**NVIDIA Driver Version**: 510.39.01

**CUDA Version**: 11.6

**CUDNN Version**: NA

Operating System: Ubuntu 18.04.6 LTS

Python Version (if applicable): 3.8.12

PyTorch Version (if applicable): 2.0.1

Baremetal or Container (if so, version): Baremetal

## Relevant Files

**Model link**:
Onnx file: [link]
Engine file: [link]
TRT-python inference script: [link]

## Steps To Reproduce
Obtain engine using onnx file attached above, if TRT version != 8.2.03.
Run trt inference on the engine attached above using script

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**: No

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`): Yes, with onnx-runtime

**Comment by zerollzeng** [comment]
Test pass in my side:
```
[I] onnxrt-runner-N0-09/01/23-14:34:03  | Completed 1 iteration(s) in 13.65 ms | Average inference time: 13.65 ms.
[I] Accuracy Comparison | trt-runner-N0-09/01/23-14:34:03 vs. onnxrt-runner-N0-09/01/23-14:34:03
[I]     Comparing Output: 'output' (dtype=float32, shape=(1, 8)) with 'output' (dtype=float32, shape=(1, 8))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-09/01/23-14:34:03: output | Stats: mean=0.125, std-dev=0.18429, var=0.033965, median=0.053548, min=0.045195 at (0, 6), max=0.61202 at (0, 4), avg-magnitude=0.125
[I]         onnxrt-runner-N0-09/01/23-14:34:03: output | Stats: mean=0.125, std-dev=0.18429, var=0.033965, median=0.053548, min=0.045195 at (0, 6), max=0.61202 at (0, 4), avg-magnitude=0.125
[I]         Error Metrics: output
[I]             Minimum Required Tolerance: elemwise error | [abs=2.3842e-07] OR [rel=8.9276e-07] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=5.0291e-08, std-dev=7.1996e-08, var=5.1834e-15, median=2.2352e-08, min=7.4506e-09 at (0, 3), max=2.3842e-07 at (0, 4), avg-magnitude=5.0291e-08
[I]             Relative Difference | Stats: mean=4.4412e-07, std-dev=2.5445e-07, var=6.4744e-14, median=3.463e-07, min=1.1478e-07 at (0, 3), max=8.9276e-07 at (0, 7), avg-magnitude=4.4412e-07
[I]         PASSED | Output: 'output' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     PASSED | All outputs matched | Outputs: ['output']
[I] Accuracy Summary | trt-runner-N0-09/01/23-14:34:03 vs. onnxrt-runner-N0-09/01/23-14:34:03 | Passed: 1/1 iterations | Pass Rate: 100.0%
[I] PASSED | Runtime: 22.406s | Command: /home/scratch.zeroz_sw/miniconda3/bin/polygraphy run resnet50.onnx --trt --onnxrt
```

pls test with the latest TRT release, because we don't fix bug on 8.2 now.

**Comment by liugangdao** [comment]
I have meet this error also in 8.4

**Comment by prashant-dn** [comment]
I use TRT8.5 container and can reproduce the similar results using `polygraphy`. But when I obtain my engine file with `--fp16` argument and then run my TRT-python script, I get similar (random) predicted classes. Then, when I remove this fp16 argument (i.e. use fp32) in `trtexec` command to obtain the engine, my python-TRT script gives great results. 

Note that when I try this (obtaining fp32 engine and inferring using python-TRT) on **TRT8.2**, the issue persists.

8.5 is the way.
